pkg/controller/provider/container/openstack/model.go

```diff
@@ -138,22 +136,17 @@ func (r *RegionAdapter) DeleteUnexisting(ctx *Context) (updates []Updater, err e
 		region := &regionList[i]
 		s := &Region{}
 		err = ctx.client.get(s, region.ID)
-		if err != nil {
-			switch err.(type) {
-			case gophercloud.ErrDefault404:
-				updater := func(tx *libmodel.Tx) (err error) {
-					m := &model.Region{
-						Base: model.Base{ID: region.ID},
-					}
-					return tx.Delete(m)
+		if err != nil && ctx.client.isNotFound(err) {
+			updater := func(tx *libmodel.Tx) (err error) {
+				m := &model.Region{
+					Base: model.Base{ID: region.ID},
 				}
-				updates = append(updates, updater)
-			default:
-				return
+				return tx.Delete(m)
 			}
+			updates = append(updates, updater)
+			err = nil
```

Comment:
what's the expectation from this method?
if we fail to get the last region due to an error that is different than NotFound, we'll return err != nil
if we fail to get a region due to an error that is different than NotFound and then got another region that is NotFound, we won't return an err = nil
I'm a bit confused here (and below :) )

Reply:
Yeah, you are right, in case we got an error in an intermediate step no error is returned... :(
I will fix it in the next push